truffle/src/main/java/org/jruby/truffle/core/string/StringNodes.java

```diff
@@ -1449,6 +1459,321 @@ private int prevCharHead(Encoding enc, byte[]bytes, int p, int s, int end) {
         }
     }
 
+    @CoreMethod(names = "scrub_internal", optional = 1, needsBlock = true)
```

Comment:
I prefer primitives over extra methods, since they are better hidden to the user.